PokemonAPI/PokemonAPI.BusinessLayer/Validator/PokemonValidator.cs

```diff
@@ -0,0 +1,107 @@
+            {
+                List<string> errors = new List<string>();
+                errors.Add("Pokemon name is required");
+                errorMap.Add("Name",errors);
+                valid = false;
+            }
+
+            if (!(Regex.IsMatch(pokemon.Name, @"^[a-zA-Z]+$")))
+            {
+                List<string> errors = new List<string>();
+                errors.Add("Pokemon name must cointain only letters");
+                errorMap.Add("Name", errors);
+                valid = false;
+            }
+
+            if (string.IsNullOrEmpty(pokemon.Avatar))
+            {
+                List<string> errors = new List<string>();
+                errors.Add("Pokemon avatar is required");
+                errorMap.Add("Avatar", errors);
+                valid = false;
+            }
+
+            if (!(Regex.IsMatch(pokemon.Avatar, @"^[a-zA-Z]+$")))
+            {
+                List<string> errors = new List<string>();
+                errors.Add("Pokemon avatar must cointain only letters");
+                errorMap.Add("Avatar", errors);
+                valid = false;
+            }
+
+            if (pokemon.Generation == null)
+            {
+                List<string> errors = new List<string>();
+                errors.Add("Pokemon generation is required");
+                errorMap.Add("Generation", errors);
+                valid = false;
+            }
+
+            if (pokemon.Generation < 1 || pokemon.Generation > 8)
+            {
+                List<string> errors = new List<string>();
+                errors.Add("Pokemon generation needs to be between 1 and 8 inclusive");
+                errorMap.Add("Generation", errors);
+                valid = false;
+            }
+            
+
+            if (pokemon.Height == 0)
+            {
+                List<string> errors = new List<string>();
+                errors.Add("Pokemon height is required");
+                errorMap.Add("Height", errors);
+                valid = false;
+            }
+
+            if (pokemon.Weight == 0)
+            {
+                List<string> errors = new List<string>();
+                errors.Add("Pokemon weight is required");
+                errorMap.Add("Weight", errors);
+                valid = false;
+            }
+
+            if (!(pokemon.Type.Equals("Normal") || pokemon.Type.Equals("Fire") || pokemon.Type.Equals("Fighting")  ||  pokemon.Type.Equals("Water") ||
+                pokemon.Type.Equals("Flying") || pokemon.Type.Equals("Grass") || pokemon.Type.Equals("Poison") || pokemon.Type.Equals("Electric") ||
+                pokemon.Type.Equals("Ground")  || pokemon.Type.Equals("Psychic") || pokemon.Type.Equals("Rock")  ||  pokemon.Type.Equals("Ice") ||
+                pokemon.Type.Equals("Bug") || pokemon.Type.Equals("Dragon") || pokemon.Type.Equals("Ghost") || pokemon.Type.Equals("Dark") ||
+                pokemon.Type.Equals("Steel") ||  pokemon.Type.Equals("Fairy")))
```

Comment:
This is too much of a place to make mistakes. Also, this doesn't need to be verified since pokemon. Type (if you look into pokemonDTO object is already of type PokemonType. If it tried to receive some other value other than **valid** Enum type it would throw an exception... ie (you wouldn't be able to make a HTTP request). This strongly typed things enables us not to worry so much about things. If we read this is type enum of PokemonType and it's not nullable we can be **sure** it's valid.

If we were to actually write this ifology
pokemon.Type == PokemonType.Normal would be more concise and less error prone without magic strings.